Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -9566,6 +9566,15 @@ void WebPage::didAdjustVisibilityWithSelectors(Vector<String>&& selectors)
     send(Messages::WebPageProxy::DidAdjustVisibilityWithSelectors(WTFMove(selectors)));
 }
 
+void WebPage::frameNameWasChangedInAnotherProcess(FrameIdentifier frameID, const String& frameName)
+{
+    RefPtr webFrame = WebProcess::singleton().webFrame(frameID);
+    if (!webFrame)
+        return;
+    if (RefPtr coreFrame = webFrame->coreFrame())
+        coreFrame->tree().setSpecifiedName(frameName.toExistingAtomString());
```

Comment:
This seems like it would fail if the AtomString doesn't already exist with those characters in it.  Can't we just call the AtomString constructor with frameName?

Reply:
Yes